src/VolumeDrawable.ts

```diff
@@ -169,33 +169,47 @@ export default class VolumeDrawable {
       if (hasIso !== options.isosurfaceEnabled) {
         if (hasIso && !options.isosurfaceEnabled) {
           this.destroyIsosurface(channelIndex);
-        } else if (!hasIso && options.isosurfaceEnabled) {
-          // 127 is half of the intensity range 0..255
-          let isovalue = 127;
-          if (options.isovalue !== undefined) {
-            isovalue = options.isovalue;
-          }
-          // 1.0 is fully opaque
-          let isosurfaceOpacity = 1.0;
-          if (options.isosurfaceOpacity !== undefined) {
-            isosurfaceOpacity = options.isosurfaceOpacity;
-          }
-          this.createIsosurface(channelIndex, isovalue, isosurfaceOpacity, isosurfaceOpacity < 1.0);
+        } else if (!hasIso && options.isosurfaceEnabled && this.volume.channels[channelIndex].loaded) {
+          const { isovalue, isosurfaceOpacity } = options;
```

Comment:
Are `isovalue` and `isosurfaceOpacity` now guaranteed to be defined?

Reply:
They now have defaults applied when `createIsosurface` is called.

...I'm realizing now that this was a choice I made before removing the specialized methods on `View3d` for updating isosurfaces. When `createIsosurface` was also called by methods on `View3d`, it made sense to keep defaults consistent at the first place where those two code paths merged. Now, that method definitely can be `private`-ized and could maybe be removed entirely (it's just one line)